plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePartitionsTable.java

```diff
@@ -0,0 +1,466 @@
+    {
+        return tableMetadata;
+    }
+
+    @Override
+    public ConnectorPageSource pageSource(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
+    {
+        if (partitionColumns.isEmpty()) {
+            return new EmptyPageSource();
+        }
+
+        return new FixedPageSource(buildPages(session));
+    }
+
+    private List<Page> buildPages(ConnectorSession session)
+    {
+        PageListBuilder pageListBuilder = PageListBuilder.forTable(tableMetadata);
+
+        Map<Map<String, Optional<String>>, DeltaLakePartitionStatistics> statisticsByPartition;
+        try (Stream<AddFileEntry> activeFiles = transactionLogAccess.loadActiveFiles(session, tableSnapshot, metadataEntry, protocolEntry, TupleDomain.all(), alwaysTrue())) {
+            statisticsByPartition = getStatisticsByPartition(activeFiles);
+        }
+
+        for (Map.Entry<Map<String, Optional<String>>, DeltaLakePartitionStatistics> partitionEntry : statisticsByPartition.entrySet()) {
+            Map<String, Optional<String>> partitionValue = partitionEntry.getKey();
+            DeltaLakePartitionStatistics deltaLakePartitionStatistics = partitionEntry.getValue();
+
+            RowType partitionValuesRowType = RowType.from(partitionFields);
+            SqlRow partitionValuesRow = buildRowValue(partitionValuesRowType, fields -> {
+                for (int i = 0; i < partitionColumns.size(); i++) {
+                    DeltaLakeColumnHandle column = partitionColumns.get(i);
+                    Type type = column.getType();
+                    Optional<String> value = partitionValue.get(column.getBasePhysicalColumnName());
+                    Object deserializedPartitionValue = deserializePartitionValue(column, value);
+                    writeNativeValue(type, fields.get(i), deserializedPartitionValue);
+                }
+            });
+
+            pageListBuilder.beginRow();
+
+            pageListBuilder.appendNativeValue(partitionValuesRowType, partitionValuesRow);
+            pageListBuilder.appendBigint(deltaLakePartitionStatistics.fileCount());
+            pageListBuilder.appendBigint(deltaLakePartitionStatistics.size());
+
+            dataColumnType.ifPresent(dataColumnType -> {
+                try {
+                    SqlRow dataColumnRow = buildRowValue(dataColumnType, fields -> {
+                        for (int i = 0; i < columnMetricTypes.size(); i++) {
+                            String fieldName = regularColumns.get(i).getBaseColumnName();
+                            Object min = deltaLakePartitionStatistics.minValues().getOrDefault(fieldName, null);
+                            Object max = deltaLakePartitionStatistics.maxValues().getOrDefault(fieldName, null);
+                            Long nullCount = deltaLakePartitionStatistics.nullCounts().getOrDefault(fieldName, null);
+                            if (min == null && max == null && nullCount == null) {
```

Comment:
Why are we disregarding the entire stats? Did you consider appending null in such cases?

Reply:
It was to maintain consistency with Iceberg connector - https://github.com/trinodb/trino/blob/master/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionTable.java#L294.

I can change it to append null per column in such cases if it makes more sense.